aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/CustomerClassLoaderTest.java

```diff
@@ -95,24 +95,35 @@ private Path fakeFileSystem(String[] paths) throws IOException {
     @Test
     @DisabledOnOs(MAC) // test fails on systems with case-insensitive volumes
     public void customerClassLoaderFunction() throws IOException {
-        Path rootDir = fakeFileSystem(EXAMPLE_FUNCTION);
-
-        URLClassLoader customerClassLoader = new CustomerClassLoader(
-                rootDir.resolve("user/path").toString(),
-                rootDir.resolve("opt/java").toString(),
-                ClassLoader.getSystemClassLoader());
-
-        List<String> res = strip("file:" + rootDir.toString(), customerClassLoader.getURLs());
-
-        Assertions.assertEquals(Arrays.asList(
-                "/user/path/",
-                "/user/path/lib/4.jar",
-                "/user/path/lib/A.jar",
-                "/user/path/lib/a.jar",
-                "/user/path/lib/b.jar",
-                "/user/path/lib/z.jar",
-                "/user/path/lib/λ.jar"),
-                res);
+        try {
+            Path rootDir = fakeFileSystem(EXAMPLE_FUNCTION);
+
+            URLClassLoader customerClassLoader = new CustomerClassLoader(
+                    rootDir.resolve("user/path").toString(),
+                    rootDir.resolve("opt/java").toString(),
+                    ClassLoader.getSystemClassLoader());
+
+            List<String> res = strip("file:" + rootDir.toString(), customerClassLoader.getURLs());
+
+            Assertions.assertEquals(Arrays.asList(
+                    "/user/path/",
+                    "/user/path/lib/4.jar",
+                    "/user/path/lib/A.jar",
+                    "/user/path/lib/a.jar",
+                    "/user/path/lib/b.jar",
+                    "/user/path/lib/z.jar",
+                    "/user/path/lib/λ.jar"),
+                    res);
+        } catch(Throwable t) {
+            // this system property is the name of the charset used when encoding/decoding file paths
+            // exception is expected if it is not set to a UTF variant
+            if (!System.getProperty("sun.jnu.encoding").toLowerCase().contains("utf")){
```

Comment:
Is this property guaranteed to be set? What happens if it's null?

Reply:
Correct. There is the possibility to be null. Treating the case in next commit.